src/Kiota.Builder/Writers/TypeScript/CodeEnumObjectWriter.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using Kiota.Builder.CodeDOM;
+using Kiota.Builder.Extensions;
+
+namespace Kiota.Builder.Writers.TypeScript;
+
+public class CodeEnumObjectWriter : BaseElementWriter<CodeEnumObject, TypeScriptConventionService>
+{
+    public CodeEnumObjectWriter(TypeScriptConventionService conventionService) : base(conventionService) { }
+    public override void WriteCodeElement(CodeEnumObject codeElement, LanguageWriter writer)
```

Comment:
I believe we can totally do this. 
However, given that the `CodeEnumObject` does end up being a `CodeElement` wouldn't it be natural to have its own writer as opposed to a having a writer that ends up writing its child elements as well?

Reply:
I'm suggesting we don't need a new Code Element type, just changing how the enum writer behaves in TypeScript should be enough

Reply:
Actually, allow me to rephrase that. I now understand you need another code element because you need to be able to reference it in other places.
My suggestion is now to use the CodeConstant code element I've introduced a couple of weeks ago since that new element is effectively a constant.
@koros would you mind refactoring your code please?

Reply:
> My suggestion is now to use the CodeConstant code element I've introduced a couple of weeks ago since that new element is effectively a constant.

+1